tests/integration/test_installation.py

```diff
@@ -166,8 +169,18 @@ def test_running_real_remove_backup_groups_job(ws, installation_ctx):
 
     installation_ctx.deployed_workflows.run_workflow("remove-workspace-local-backup-groups")
 
-    with pytest.raises(NotFound):
-        ws.groups.get(ws_group_a.id)
+    # It takes a moment to delete the group through the API
+    start_time = time.time()
+    while time.time() - start_time < timeout_in_seconds:
+        try:
+            ws.groups.get(ws_group_a.id)
+        except NotFound:
+            assert True, "Group removed"
+            break
+        else:
+            time.sleep(seconds_inbetween_tries)
+    else:
+        assert False, "Group not removed"
```

Comment:
```suggestion
    @reried(on=[NotFound], timeout=timedelta(seconds=60))
    def wait():
        ws.groups.get(ws_group_a.id)
    wait()
```
it's a bit less verbose and uses existing frameworks